app/scenes/ApiKeyNew/components/ExpiryDatePicker.tsx

```diff
@@ -0,0 +1,116 @@
+import { faCalendar } from "@fortawesome/free-solid-svg-icons";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import { isToday, isPast, format as formatDate } from "date-fns";
+import React from "react";
+import { DayPicker } from "react-day-picker";
+import { useTranslation } from "react-i18next";
+import { usePopoverState, PopoverDisclosure } from "reakit";
+import styled, { useTheme } from "styled-components";
+import { dateLocale } from "@shared/utils/date";
+import Button from "~/components/Button";
+import Flex from "~/components/Flex";
+import Popover from "~/components/Popover";
+import Text from "~/components/Text";
+import useUserLocale from "~/hooks/useUserLocale";
+
+type Props = {
+  selectedDate?: Date;
+  onSelect: (date: Date) => void;
+};
+
+const ExpiryDatePicker = ({ selectedDate, onSelect }: Props) => {
+  const { t } = useTranslation();
+  const theme = useTheme();
+
+  const userLocale = useUserLocale();
+  const locale = React.useMemo(() => dateLocale(userLocale), [userLocale]);
```

Comment:
```suggestion
  const locale = dateLocale(userLocale);
```

No memo needed it's a simple lookup